### PR TITLE
[libpng] fix Android ARM triplet build failure

### PR DESCRIPTION
**Pull request by luncliff**
### What does your PR fix?  

Fix build failure of `libpng` targeting Android `arm-android`.

Port version 14 fails with the following error (install-arm-android-dbg-out.log)

```
...
[6/21] D:\shared\Android\Sdk\ndk\21.4.7075529\toolchains\llvm\prebuilt\windows-x86_64\bin\clang.exe -target armv7-none-linux-androideabi21 -gcc-toolchain D:/shared/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/windows-x86_64 --sysroot=D:/shared/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/windows-x86_64/sysroot -DPNG_ARM_NEON_OPT=2 -I. -ID:/luncliff/sources/vcpkg/buildtrees/libpng/src/v1.6.37-05658d6f1a.clean -isystem D:/luncliff/sources/vcpkg/installed/arm-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -mfpu=vfpv3-d16 -Wformat -Werror=format-security  -O0 -fno-limit-debug-info  -fPIC -MD -MT CMakeFiles/png_static.dir/arm/filter_neon.S.o -MF CMakeFiles\png_static.dir\arm\filter_neon.S.o.d -o CMakeFiles/png_static.dir/arm/filter_neon.S.o -c D:/luncliff/sources/vcpkg/buildtrees/libpng/src/v1.6.37-05658d6f1a.clean/arm/filter_neon.S
FAILED: CMakeFiles/png_static.dir/arm/filter_neon.S.o 
D:\shared\Android\Sdk\ndk\21.4.7075529\toolchains\llvm\prebuilt\windows-x86_64\bin\clang.exe -target armv7-none-linux-androideabi21 -gcc-toolchain D:/shared/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/windows-x86_64 --sysroot=D:/shared/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/windows-x86_64/sysroot -DPNG_ARM_NEON_OPT=2 -I. -ID:/luncliff/sources/vcpkg/buildtrees/libpng/src/v1.6.37-05658d6f1a.clean -isystem D:/luncliff/sources/vcpkg/installed/arm-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -mfpu=vfpv3-d16 -Wformat -Werror=format-security  -O0 -fno-limit-debug-info  -fPIC -MD -MT CMakeFiles/png_static.dir/arm/filter_neon.S.o -MF CMakeFiles\png_static.dir\arm\filter_neon.S.o.d -o CMakeFiles/png_static.dir/arm/filter_neon.S.o -c D:/luncliff/sources/vcpkg/buildtrees/libpng/src/v1.6.37-05658d6f1a.clean/arm/filter_neon.S
<instantiation>:18:9: error: unknown directive
        .func png_read_filter_row_sub4_neon
        ^
...
```

Change option `PNG_ARM_NEON=on` to `PNG_ARM_NEON=check`.

#### Related Issues

* #16847 current mainstream's `arm-android.cmake`
* #11162 `PNG_ARM_NEON=on` specified

### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  

* `arm-android`

### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  

Yes

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Waiting for https://github.com/microsoft/vcpkg/pull/18116 merge.

**Comment by luncliff** [comment]
I found `arm64-osx` triplet build is failing. Let me handle the issue in this PR.

File: ports/libpng/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "libpng",
+  "version-string": "1.6.37",
```
Comment (JackBoosY) [inline ports/libpng/vcpkg.json:3]:
```suggestion
  "version": "1.6.37",
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
Please ping me if this PR is ready for review.

**Comment by luncliff** [comment]
I checked triplets and build was successful with local machine. I think it will be fine when CI passes.

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
LGTM.

**Comment by strega-nil-ms** [comment]
Thanks @luncliff :)

**Pull request merged**